gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -131,19 +130,17 @@ public AbstractJobLauncher(Properties properties)
   public void launchJob(Properties jobProps, JobListener jobListener)
       throws JobException {
     Preconditions.checkNotNull(jobProps);
+    Preconditions.checkArgument(
+        jobProps.containsKey(ConfigurationKeys.JOB_NAME_KEY), "A job must have a job name specified by job.name");
 
     String jobName = jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY);
-    if (Strings.isNullOrEmpty(jobName)) {
-      throw new JobException("A job must have a job name specified by job.name");
-    }
 
-    String jobDisabled = jobProps.getProperty(ConfigurationKeys.JOB_DISABLED_KEY, "false");
-    if (Boolean.valueOf(jobDisabled)) {
+    if (Boolean.valueOf(jobProps.getProperty(ConfigurationKeys.JOB_DISABLED_KEY, Boolean.FALSE.toString()))) {
```

Comment:
nit: Please don't read properties from the configuration outside the constructor, especially if there are default values. Please add a jobDisabled member field.